### PR TITLE
Add Windows + CUDA

### PR DESCRIPTION
**Pull request by sodre**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,9 +52,12 @@ requirements:
     - git
     - ninja
   host:
-    - libcublas-dev  # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-cudart-dev  # [(cuda_compiler_version or "").startswith("12")]
+    - libcublas-dev    # [(cuda_compiler_version or "").startswith("12")]
   run:
     - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+    - cuda-cudart  {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-cudart_{{ target_platform }} {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12") and win]
```
Comment (sodre) [inline recipe/meta.yaml:60]: Something Is wrong with the Windows dependencies. Unless I add the line above I get the following error on Windows

```
  ERROR (llama.cpp,bin/main.exe): Needed DSO Library/bin/cudart64_12.dll found in ['cuda-cudart_win-64']
  ERROR (llama.cpp,bin/main.exe): .. but ['cuda-cudart_win-64'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

**Review by sodre — COMMENTED** [review]
Document that cuda-cudart_win-64 package is not populating automatically

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:60]: Yeah this is due to upstream issue ( https://github.com/conda-forge/cuda-cudart-feedstock/issues/13#event-11242601478 )

We just merged a fix ( https://github.com/conda-forge/cuda-cudart-feedstock/pull/14 ). So this should be resolved with the new packages

Think we can drop this then

Comment (jakirkham) [reply recipe/meta.yaml:60]: Raised issue ( https://github.com/conda-forge/llama.cpp-feedstock/issues/16 ) to track

**Pull request merged**